checkov/common/util/json_utils.py

```diff
@@ -30,9 +34,62 @@ def default(self, o: Any) -> Any:
             return o.__dict__
         elif isinstance(o, type_of_function):
             return str(o)
+        elif isinstance(o, TFDefinitionKey):
+            return str(o)
+        elif isinstance(o, TFModule):
+            return str(o)
         else:
             return json.JSONEncoder.default(self, o)
 
+    def encode(self, obj):
+        return super().encode(self._encode(obj))
+
+    def _encode(self, obj):
+        if isinstance(obj, dict):
+            return {self.encode_key(k): v for k, v in obj.items()}
+        else:
+            return obj
+
+    @staticmethod
+    def encode_key(key):
+        if isinstance(key, TFDefinitionKey):
+            return str(key)
+        if isinstance(key, TFModule):
+            return str(key)
+        else:
+            return key
+
+
+def object_hook(dct):
+    try:
+        if dct is None:
+            return None
+        if isinstance(dct, dict):
+            dct_obj = deepcopy(dct)
+            if 'tf_source_modules' in dct and 'file_path' in dct:
+                return TFDefinitionKey(file_path=dct["file_path"],
+                                       tf_source_modules=object_hook(dct["tf_source_modules"]))
+            if 'path' in dct and 'name' in dct and 'foreach_idx' in dct and 'nested_tf_module' in dct:
```

Comment:
are any of this saved consts? if not you don't have to refactor them as they are used only once, but I figured they might be already defined

Reply:
Nope